File: android/src/main/java/com/fantastic/manage_calendar_events/models/Calendar.java
package com.fantastic.manage_calendar_events.models;
import com.google.gson.annotations.SerializedName;

public final class Calendar {

  @SerializedName("id")
  private final String id;
  @SerializedName("name")
  private final String name;
  @SerializedName("accountName")
  private final String accountName;
  @SerializedName("ownerName")
  private final String ownerName;

  public Calendar(String id, String name, String accountName, String ownerName) {
    this.id = id;
    this.name = name;
    this.accountName = name;
    this.ownerName = name;
  }

  public String getId() {
    return id;
  }

  public String getName() {
    return name;
  }

  public String getAccountName() {
    return accountName;
  }

  public String getOwnerName() {
    return ownerName;
  }

  @Override
  public String toString() {
    return new StringBuffer().append(id).append("-").append(name).append("-").append(accountName)
        .append("-")
        .append(ownerName).toString();
  }
}

File: lib/src/model/calendar.dart
part of manage_calendar_events;

class Calendar {
  String? id;
  String? name;
  String? accountName;
  String? ownerName;
  bool? isReadOnly;

  Calendar({
    required this.id,
    this.name,
    this.accountName,
    this.ownerName,
    this.isReadOnly,
  });

  Calendar.fromJson(Map<String, dynamic> data) {
    this.id = data["id"];
    this.name = data["name"];
    this.accountName = data["accountName"];
    this.ownerName = data["ownerName"];
    this.isReadOnly = data["isReadOnly"];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = Map<String, dynamic>();
    data["id"] = this.id;
    data["name"] = this.name;
    data["accountName"] = this.accountName;
    data["ownerName"] = this.ownerName;
    data["isReadOnly"] = this.isReadOnly;
    return data;
  }
}
